core/modules/node/node.module

```diff
@@ -1901,6 +1909,33 @@ function node_block_view($delta = '', $settings = array()) {
         }
       }
       break;
+
+    case 'empty':
+      // Check if the promoted view exists and still has the block display.
+      $view = views_get_view('promoted');
+      if ($view && $view->access('block')) {
+        $nids = views_get_view_result('promoted', 'block');
+
+        if (empty($nids)) {
+          $default_message = '<p>' . t('No content has been promoted to the front page yet.') . '</p>';
```

Comment:
I think we've nixed "Promote to front page" in preference of just "Promote" on the node forms. As you could use this promoted view other places besides the front page, I think it makes sense that we nix "to the front page" and just say "No content has been promoted yet."